### DISCUSSION TITLE
A better way to merge arguments 'key' and 'onTap' of class <PaneItem>?

### DISCUSSION THREAD

**Discussion by AClon314** [bdlukaa/fluent_ui#938] [Ideas]
Hello, @bdlukaa 
## Modify from Official fluent_ui/example repo
[from `fluent_ui/example/lib/main.dart`: `class _MyHomePageState extends State<MyHomePage> with WindowListener {`](https://github.com/bdlukaa/fluent_ui/blob/abb4df620c443cb4fba4fa639bfe0b0eecf41995/example/lib/main.dart#L152C12-L152C12)
Original:
```dart
late final List<NavigationPaneItem> originalItems = [
    PaneItem(
      key: const ValueKey('/'),
      icon: const Icon(FluentIcons.home),
      title: const Text('Home'),
      body: const SizedBox.shrink(),
    ),
```
Modified:
```dart
setOnTap(String path) {
    if (GoRouterState.of(context).uri.toString() != path) {
      context.go(path);
    }
}

late final List<NavigationPaneItem> originalItems = [
    PaneItem(
      key: const ValueKey('/'),
      onTap: () => setOnTap('/'),
      // but I don't want to write the path twice, in this example it's '/', but could be longer like '/pathA/pathB/pathC/...'
      icon: const Icon(FluentIcons.home),
      title: const Text('Home'),
      body: const SizedBox.shrink(),
    ),
```
[this could avoid the **remaping** `].map((e) { if (e is PaneItem) { return PaneItem(...) }).toList();`](https://github.com/bdlukaa/fluent_ui/blob/abb4df620c443cb4fba4fa639bfe0b0eecf41995/example/lib/main.dart#L334),which would cause Class `<PaneItemExpander>` loose the property `items`,`initiallyExpanded`...

Also make the `onTap` more customized.

### My Suggestion
make `{void Function()? onTap}` to `{void Function(key)? onTap}`, so we could write like this:
```dart
late final List<NavigationPaneItem> originalItems = [
    PaneItem(
      key: const ValueKey('/'),
      onTap: (key) => setOnTap(key), //or (_)=>myFunction()
      icon: const Icon(FluentIcons.home),
      title: const Text('Home'),
      body: const SizedBox.shrink(),
    ),
```

I'm busy with my tiny demo project, so I write my suggestion first here, later I would try to pull request to solve this maybe :)

**Comment by bdlukaa** [comment]
I like the idea, but I think it'd create one more field we'd need to edit if we make any changes to the pane items - and we don't make use of `PaneItemExpander` in the example app anyways.

Though, the example app is just an example and not a model of how every application should be. In fact, with relatively small apps, there isn't the need to make use of `go_router` at all.
